update Compose release notes page for v2.20.0 (#17720)
docker/docs: compose/release-notes.md

> For more detailed information, see the [release notes in the Compose repo](https://github.com/docker/compose/releases/tag/v2.18.0).

Reviewer: ```suggestion
For more detailed information, see the [release notes in the Compose repo](https://github.com/docker/compose/releases/).
```

## 2.20.0
{% include release-date.html date="2023-07-11" %}

### Update
- Dependencies upgrade: bump docker/cli-docs-tools to v0.6.0
- Dependencies upgrade: bump docker to v24.0.4
- Dependencies upgrade: bump buildx to v0.11.1

### Bug fixes and enhancements

> * Introduce `wait` command

Reviewer: ```suggestion
* Introduced the `wait` command.
```

> * Add support of `--builder` and `BUILDX_BUILDER` to `build` command

Reviewer: ```suggestion
* Added support of `--builder` and `BUILDX_BUILDER` to the `build` command.
```

> * Support `attach` attribute from the Compose Specification

Reviewer: ```suggestion
* Added support for the `attach` attribute from the Compose Specification.
```

> * Fixed a DryRun mode issue when initializing CLI client

Reviewer: ```suggestion
* Fixed a DryRun mode issue when initializing CLI client.
```

> * Fixed a bug with random missing network when service has more than one

Reviewer: ```suggestion
* Fixed a bug with random missing network when a service has more than one.
```

> * Fixed the Secrets file permission value to comply with spec

Reviewer: ```suggestion
* Fixed the Secrets file permission value to comply with the Compose Specification.
```

> * Fixed an issue about `no-deps` flag not applied 

Reviewer: ```suggestion
* Fixed an issue about `no-deps` flag not being applied. 
```

> * Fixed some source code comments

Reviewer: ```suggestion
* Fixed some source code comments.
```

> * Fixed a bug when `--index` is not set select

Reviewer: ```suggestion
* Fixed a bug when `--index` is not set select.
```

> * Fix a process leak in wait e2e test

Reviewer: ```suggestion
* Fixed a process leak in the wait e2e test.
```

> * Improve some tests speed

Reviewer: ```suggestion
* Improved some test speeds.
```
